### PR TITLE
Add fast user resource, closes #1092

### LINKED ISSUE #1092: implement Fast User resource

### PR DESCRIPTION
**Pull request by briankariuki**
This resource is similar to the user resource, but it loads all calculations (city, roles, profile_photo) and aggregates (credit_points) in one sql query.

### CODE REVIEW

File: lib/animina/accounts/resources/fast_user.ex [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  This is the Fast User module.
+  """
+
+  use Ash.Resource,
+    data_layer: AshPostgres.DataLayer,
+    authorizers: Ash.Policy.Authorizer,
+    domain: Animina.Accounts,
+    extensions: []
+
+  postgres do
```
Comment (wintermeyer) [inline lib/animina/accounts/resources/fast_user.ex:23]: Is it possible to set this read only? If yes we should.

File: lib/animina/accounts/resources/fast_user.ex [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+      run fn input, _ ->
+        {:ok, []}
+      end
+    end
+  end
+
+  attributes do
+    uuid_primary_key :id
+    attribute :email, :ci_string, allow_nil?: false, public?: true
+    attribute :hashed_password, :string, allow_nil?: false, sensitive?: true
```
Comment (wintermeyer) [inline lib/animina/accounts/resources/fast_user.ex]: Let's keep this to the minimum which we need for the profile view. Which means e.g. no need for the `:hashed_password` and other attributes. Strip it down as fare as possible.

File: lib/animina/accounts/resources/fast_user.ex [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    attribute :is_in_waitlist, :boolean, default: false, public?: true
+
+    attribute :username, :ci_string do
+      allow_nil? false
```
Comment (wintermeyer) [inline lib/animina/accounts/resources/fast_user.ex]: If we make the resource read-only we don't need the constraints or validations.

**Review by wintermeyer — CHANGES_REQUESTED** [review]

File: lib/animina/accounts/resources/fast_user.ex (hunk 1 above)
Comment (briankariuki) [reply lib/animina/accounts/resources/fast_user.ex:23]: I couldn't find any documentation on how to make a resource read only. FastUser only supports the read action

**Review by wintermeyer — APPROVED** [review]
Nice work @briankariuki !

Maybe we need a bit more documentation for other developers to understand the code but this is a major speed improvement. I like it.

**Comment by briankariuki** [comment]
> Nice work @briankariuki !
> 
> Maybe we need a bit more documentation for other developers to understand the code but this is a major speed improvement. I like it.

I'll add explanations in the discussion tab later in the week

**Pull request merged**